Patch proposed by Martin Lottermoser for apt.systemd.daily in apt 2.6.1.

--- apt.systemd.daily.orig	2023-05-25 16:11:37.000000000 +0200
+++ apt.systemd.daily	2023-08-28 08:29:06.874803852 +0200
@@ -102,25 +102,14 @@
         return 0
     fi
 
-    # compare midnight today to midnight the day the stamp was updated
     stamp_file="$stamp"
-    stamp=$(date --date="$(date -r "$stamp_file" --iso-8601)" +%s 2>/dev/null)
+    stamp=$(date -r "$stamp_file" +%s 2>/dev/null)
     if [ "$?" != "0" ]; then
-        # Due to some timezones returning 'invalid date' for midnight on
-        # certain dates (e.g. America/Sao_Paulo), if date returns with error
-        # remove the stamp file and return 0. See coreutils bug:
-        # http://lists.gnu.org/archive/html/bug-coreutils/2007-09/msg00176.html
         rm -f "$stamp_file"
         return 0
     fi
 
-    now=$(date --date="$(date --iso-8601)" +%s 2>/dev/null)
-    if [ "$?" != "0" ]; then
-        # As above, due to some timezones returning 'invalid date' for midnight
-        # on certain dates (e.g. America/Sao_Paulo), if date returns with error
-        # return 0.
-        return 0
-    fi
+    # ${now} has already been set in the main section.
 
     delta=$((now-stamp))
 
@@ -200,8 +189,7 @@
 	# maxSize is in MB
 	MaxSize=$((MaxSize*1024))
 
-	#get current time
-	now=$(date --date="$(date --iso-8601)" +%s)
+	# MinAge is in days
 	MinAge=$((MinAge*24*60*60))
 
 	# reverse-sort by mtime
@@ -219,6 +207,7 @@
 		# check both ctime and mtime 
 		mtime=$(stat -c %Y "$file")
 		ctime=$(stat -c %Z "$file")
+		# ${now} has already been set in the main section.
 		if [ "$mtime" -gt "$ctime" ]; then
 		    delta=$((now-mtime))
 		else
@@ -388,6 +377,10 @@
 
 # Global current time in seconds since 1970-01-01 00:00:00 UTC
 now=$(date +%s)
+if [ $? -ne 0 ]; then
+    echo "E: Cannot determine current time" >&2
+    exit 1
+fi
 
 # Support old Archive for compatibility.
 # Document only Periodic for all controlling parameters of this script.
